src/components/Explore.tsx

```diff
@@ -0,0 +1,263 @@
+import { PiLightningFill, PiSwordFill, PiAlienFill } from "react-icons/pi";
+import { FaLaugh } from "react-icons/fa";
+import { FaHourglassHalf, FaWandMagicSparkles } from "react-icons/fa6";
+import { GiMagnifyingGlass, GiSpiderWeb } from "react-icons/gi";
+import { IoSunnySharp } from "react-icons/io5";
+import { Field, Form, Formik } from "formik";
+import * as Yup from "yup";
+import { useState } from "react";
+import { searchComicByGenre, searchComicByName } from "../services/api";
+import { useQuery } from "@tanstack/react-query";
+import { SearchComicContent } from "../models/state";
+import ComicPopup from "./ComicPopup";
+
+function Explore() {
+  const [selectedGenre, setSelectedGenre] = useState("");
+  const [searchedComicName, setSearchedComicName] = useState("");
+
+  // Validation for input string
+  const comicSearchSchema = Yup.object().shape({
+    searchString: Yup.string()
+      .min(2, "The input is too short.")
+      .max(150, "The input is too long.")
+      .required("Required."),
+  });
+
+  // Some popular genres to search for
+  const comicGenres = [
+    "Action",
+    "Adventure",
+    "Comedy",
+    "Historical",
+    "Magic",
+    "Mystery",
+    "Sci-Fi",
+    "Slice of Life",
+  ];
+
+  // Used to set the background of the respective pills associated with each genre.
+  const colourCodeGenreLabels = [
+    "bg-red-600/20",
+    "bg-orange-500/20",
+    "bg-yellow-500/20",
+    "bg-lime-500/20",
+    "bg-cyan-500/20",
+    "bg-purple-400/20",
+    "bg-green-700/20",
+    "bg-pink-500/20",
+  ];
+
+  // Icons assoicated with each genre.
+  const comicGenreIcons = [
+    <PiLightningFill className="text-red-600 text-lg" />,
+    <PiSwordFill className="text-orange-500 text-lg" />,
+    <FaLaugh className="text-yellow-500 text-lg" />,
+    <FaHourglassHalf className="text-lime-500 text-lg" />,
+    <FaWandMagicSparkles className="text-cyan-500 text-lg" />,
+    <GiMagnifyingGlass className="text-purple-400 text-lg" />,
+    <PiAlienFill className="text-green-700 text-lg" />,
+    <IoSunnySharp className="text-pink-500 text-lg" />,
+  ];
```

Comment:
A suggestion, maybe large mapping data, such as the three above could be in a different file. That way, it doesn't cause clutter in this one component file.